### PR TITLE
docs: add NVIDIA docs

### PR DESCRIPTION
**Pull request by frezbo**
Add NVIDIA docs

Signed-off-by: Noel Georgi <git@frezbo.dev>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5230)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -192,6 +192,9 @@ docs: ## Generates the documentation for machine config, and talosctl.
 	@rm -rf docs/talosctl/*
 	@$(MAKE) local-$@ DEST=./ PLATFORM=linux/amd64
 
+docs-preview: ## Starts a local preview of the documentation using Hugo
```
Comment (smira) [inline Makefile:195]: nit: we should make it `.PHONY` to signal that it doesn't generate anything actually )

File: website/content/v1.0/guides/nvidia-gpu.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+This guide assumes the user has access to a container registry with push permissions, docker installer on the build machine and the Talos host has access to the registry.
+
+Set the local registry and username environment variables:
+
+```bash
+export USERNAME=<username>
+export REGISTRY=<registry>
+export TAG=<custom installer image tag>
+```
```
Comment (smira) [inline website/content/v1.0/guides/nvidia-gpu.md:27]: probably it makes sense to give some examples and take a note that one should be able to push to that registry (have credentials/auth)

File: website/content/v1.0/guides/nvidia-gpu.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+```Dockerfile
+FROM scratch as customization
+COPY --from=<registry>/<username>/nonfree-kmod-nvidia:<custom installer image tag> /lib/modules /lib/modules
+
+FROM ghcr.io/siderolabs/installer:{{< release >}}
+COPY --from=<registry>/<username>/kernel:<custom installer image tag> /boot/vmlinuz /usr/install/${TARGETARCH}/vmlinuz
+```
+
+Now built the image and push it to the registry.
```
Comment (smira) [inline website/content/v1.0/guides/nvidia-gpu.md:53]:
```suggestion
Now build the image and push it to the registry.
```

File: website/content/v1.0/guides/nvidia-gpu.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+Now run the following command to build and push the system extension.
+
+Replace the platform with `linux/arm64` if building for ARM64
+
+```bash
+make nvidia-container-toolkit PLATFORM=linux/amd64 PUSH=true
+```
+
+## Upgrading Talos and enabling the NVIDIA modules and the system extension
```
Comment (smira) [inline website/content/v1.0/guides/nvidia-gpu.md:74]: should we explain that first we need to deploy a Talos cluster first?

File: website/content/v1.0/guides/nvidia-gpu.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+      - name: nvidia_uvm
+      - name: nvidia_drm
+      - name: nvidia_modeset
+- op: add
+  path: /machine/sysctls
+  value:
+    net.core.bpf_jit_harden: 1
+```
+
+Now apply the patch to the machine config:
```
Comment (smira) [inline website/content/v1.0/guides/nvidia-gpu.md:99]: probably some note that we should do it for all (worker) nodes?

File: website/content/v1.0/guides/nvidia-gpu.md (hunk 2 above)
Comment (frezbo) [reply website/content/v1.0/guides/nvidia-gpu.md:27]: The auth part is mentioned here https://github.com/siderolabs/talos/pull/5230/files#diff-612c3f5d62e58a91892944466ccb1871e565255ee8b3c97e5487d3a742adfc17R19

File: Makefile (hunk 1 above)
Comment (frezbo) [reply Makefile:196]: fixed

File: website/content/v1.0/guides/nvidia-gpu.md (hunk 2 above)
Comment (frezbo) [reply website/content/v1.0/guides/nvidia-gpu.md:27]: fixed

**Review by smira — APPROVED** [review]
:cool:

**Pull request merged**